src/main.rs

```diff
@@ -325,3 +338,72 @@ pub fn check_arguments(targets: &HashSet<Target>, espidf_version: &Option<String
 
     Ok(())
 }
+
+pub fn get_host_triple(host_triple_arg: Option<String>) -> Result<String, String> {
+    match host_triple_arg {
+        Some(host_triple) => {
+            if !matches!(
+                host_triple.as_str(),
+                "x86_64-unknown-linux-gnu"
+                    | "aarch64-unknown-linux-gnu"
+                    | "x86_64-pc-windows-msvc"
+                    | "x86_64-pc-windows-gnu"
+                    | "x86_64-apple-darwin"
+                    | "aarch64-apple-darwin"
```

Comment:
Might be a good idea to pull these out into a constant.

Reply:
Yes, maybe use store them in an `enum`? It could be reused in other places like [here](https://github.com/esp-rs/espup/blob/main/src/toolchain/llvm_toolchain.rs#L31-L50) and a few other places where we rewrite those targets.

Reply:
Just updated it! I've created [`host_triple.rs`](https://github.com/esp-rs/espup/blob/feature/default-host-arg/src/host_triple.rs) which is very similar to `targets.rs`, and changed all the places where we were matching host triples.